src/Files.App/ViewModels/UserControls/Previews/ShellPreviewViewModel.cs

```diff
@@ -161,61 +161,61 @@ private unsafe bool ChildWindowToXaml(nint parent, UIElement presenter)
 				D3D_DRIVER_TYPE.D3D_DRIVER_TYPE_WARP,
 			];
 
-			ID3D11Device* pD3D11Device = default;
-			ID3D11DeviceContext* pD3D11DeviceContext = default;
-
-			foreach (var driveType in driverTypes)
+			Windows.Win32.Foundation.HRESULT hr = default;
+			Guid IID_IDCompositionDevice = typeof(IDCompositionDevice).GUID;
+			using ComPtr<ID3D11Device> pD3D11Device = default;
+			using ComPtr<ID3D11DeviceContext> pD3D11DeviceContext = default;
+			using ComPtr<IDXGIDevice> pDXGIDevice = default;
+			using ComPtr<IDCompositionDevice> pDCompositionDevice = default;
+			using ComPtr<IUnknown> pControlSurface = default;
+			ComPtr<IDCompositionVisual> pChildVisual = default; // Don't dispose this one, it's used by the compositor
```

Comment:
It seems that we never release it and would cause memory leaking. I think we should do something around this to free it eventually when user switched or closed the preview pane.